`jacocoMergedReport` task fails on Gradle 8 when invoked after `test` task

### Description

The `jacocoMergedReport` task fails to run on Gradle 8 due to a deprecated behaviour that is considered an error in this version. On Gradle 7 the same sequence of tasks shows the deprecation warning:

```console
$ ./gradlew --version

------------------------------------------------------------
Gradle 7.6.2
------------------------------------------------------------
...
$ ./gradlew --warn clean build jacocoMergedReport
...
Execution optimizations have been disabled for task ':jacocoMergedReport' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/home/xxx/ca-project'. Reason: Task ':jacocoMergedReport' uses this output of task ':test' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.6.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
...
```

### Reproduce

```console
$ mkdir ca-project
$ cd ca-project
$ gradle init --dsl groovy  --package com.example --type java-application --project-name ca-project  --test-framework junit-jupiter
$ ./gradlew wrapper --gradle-version 8.2.1
$ echo "plugins {
    id \"co.com.bancolombia.cleanArchitecture\" version \"3.6.0\"
}" > build.gradle
$ ./gradlew ca --coverage JACOCO --example TRUE --javaVersion VERSION_17 --lombok TRUE --metrics FALSE --name ca-project --package com.example --type IMPERATIVE
$ ./gradlew --warn clean build jacocoMergedReport
...
FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':jacocoMergedReport' (type 'JacocoReport').
  - Gradle detected a problem with the following location: '/home/xxx/ca-project'.
    
    Reason: Task ':jacocoMergedReport' uses this output of task ':test' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':test' as an input of ':jacocoMergedReport'.
      2. Declare an explicit dependency on ':test' from ':jacocoMergedReport' using Task#dependsOn.
      3. Declare an explicit dependency on ':test' from ':jacocoMergedReport' using Task#mustRunAfter.
    
    For more information, please refer to https://docs.gradle.org/8.2.1/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.
...
```

### Additional info

It seems the merged report execution data creates a dependency on the file `ca-project/build/jacoco/test.exec`, which is an output of the `test` task but is never created. It's included because the wildcard `**` matches the project root directory.

https://github.com/bancolombia/scaffold-clean-architecture/blob/7426c128b708f489f28290203be4a44519c625b1/src/main/resources/structure/root/main.gradle.mustache#L88

--- a/src/main/java/co/com/bancolombia/factory/upgrades/actions/UpgradeY2023M10D02JacocoMerged.java
+++ b/src/main/java/co/com/bancolombia/factory/upgrades/actions/UpgradeY2023M10D02JacocoMerged.java
@@ -1,0 +1,27 @@
+package co.com.bancolombia.factory.upgrades.actions;
+
+import co.com.bancolombia.Constants;
+import co.com.bancolombia.factory.ModuleBuilder;
+import co.com.bancolombia.factory.upgrades.UpgradeAction;
+import lombok.SneakyThrows;
+
+public class UpgradeY2023M10D02JacocoMerged implements UpgradeAction {
+  public static final String BEFORE = "dependsOn = subprojects.jacocoTestReport";
+  public static final String AFTER = "dependsOn = [test, subprojects.jacocoTestReport]";
+
+  @Override
+  @SneakyThrows
+  public boolean up(ModuleBuilder builder) {
+    return builder.updateExpression(Constants.MainFiles.MAIN_GRADLE, BEFORE, AFTER);
+  }
+
+  @Override
+  public String name() {
+    return "3.6.1->3.6.2";
+  }
+
+  @Override
+  public String description() {
+    return "Add jacocoMergedReport dependency on test";
+  }
+}

--- a/src/main/resources/structure/root/main.gradle.mustache
+++ b/src/main/resources/structure/root/main.gradle.mustache
@@ -81,7 +81,7 @@ jacoco {
 }
 
 tasks.register('jacocoMergedReport', JacocoReport) {
-    dependsOn = subprojects.jacocoTestReport
+    dependsOn = [test, subprojects.jacocoTestReport]
     additionalSourceDirs.setFrom files(subprojects.sourceSets.main.allSource.srcDirs)
     sourceDirectories.setFrom files(subprojects.sourceSets.main.allSource.srcDirs)
     classDirectories.setFrom files(subprojects.sourceSets.main.output)

--- a/src/test/java/co/com/bancolombia/factory/upgrades/actions/UpgradeY2023M10D02JacocoMergedTest.java
+++ b/src/test/java/co/com/bancolombia/factory/upgrades/actions/UpgradeY2023M10D02JacocoMergedTest.java
@@ -1,0 +1,54 @@
+package co.com.bancolombia.factory.upgrades.actions;
+
+import static org.gradle.internal.impldep.org.testng.Assert.assertNotNull;
+import static org.gradle.internal.impldep.org.testng.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+import co.com.bancolombia.Constants;
+import co.com.bancolombia.factory.ModuleBuilder;
+import co.com.bancolombia.factory.upgrades.UpgradeAction;
+import co.com.bancolombia.utils.FileUtils;
+import com.github.mustachejava.resolver.DefaultResolver;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UpgradeY2023M10D02JacocoMergedTest {
+  @Mock private Project project;
+  @Mock private Logger logger;
+  private ModuleBuilder builder;
+  private UpgradeAction updater;
+
+  @Before
+  public void setup() throws IOException {
+    when(project.getName()).thenReturn("UtilsTest");
+    when(project.getLogger()).thenReturn(logger);
+    when(project.getProjectDir()).thenReturn(Files.createTempDirectory("sample").toFile());
+    builder = spy(new ModuleBuilder(project));
+    updater = new UpgradeY2023M10D02JacocoMerged();
+    assertNotNull(updater.name());
+    assertNotNull(updater.description());
+  }
+
+  @Test
+  public void shouldApplyUpdate() throws IOException {
+    String file = Constants.MainFiles.MAIN_GRADLE;
+    // Arrange
+    DefaultResolver resolver = new DefaultResolver();
+    String text = FileUtils.getResourceAsString(resolver, "jacoco-merged/before.txt");
+    String expectedText = FileUtils.getResourceAsString(resolver, "jacoco-merged/after.txt");
+    builder.addFile(file, text);
+    // Act
+    boolean applied = updater.up(builder);
+    // Assert
+    assertTrue(applied);
+    verify(builder, times(1)).addFile(file, expectedText);
+  }
+}

--- a/src/test/resources/jacoco-merged/after.txt
+++ b/src/test/resources/jacoco-merged/after.txt
@@ -1,0 +1,68 @@
+allprojects {
+    repositories {
+        mavenCentral()
+        maven { url "https://repo.spring.io/snapshot" }
+        maven { url "https://repo.spring.io/milestone" }
+    }
+}
+
+subprojects {
+    apply plugin: 'java'
+    apply plugin: 'jacoco'
+    apply plugin: 'io.spring.dependency-management'
+
+    compileJava.dependsOn validateStructure
+    sourceCompatibility = JavaVersion.VERSION_17
+    test {
+        useJUnitPlatform()
+    }
+    dependencies {
+        testImplementation 'org.springframework.boot:spring-boot-starter-test'
+        compileOnly "org.projectlombok:lombok:${lombokVersion}"
+        annotationProcessor  "org.projectlombok:lombok:${lombokVersion}"
+        testCompileOnly  "org.projectlombok:lombok:${lombokVersion}"
+        testAnnotationProcessor  "org.projectlombok:lombok:${lombokVersion}"
+        implementation platform("org.springframework.boot:spring-boot-dependencies:${springBootVersion}")
+    }
+
+
+    test.finalizedBy(project.tasks.jacocoTestReport)
+
+    jacocoTestReport {
+        dependsOn test
+        reports {
+            xml.setRequired true
+            xml.setOutputLocation file("${buildDir}/reports/jacoco.xml")
+            csv.setRequired false
+            html.setOutputLocation file("${buildDir}/reports/jacocoHtml")
+        }
+    }
+
+}
+
+jacoco {
+    toolVersion = "${jacocoVersion}"
+    reportsDirectory.set(file("$buildDir/reports"))
+}
+
+tasks.register('jacocoMergedReport', JacocoReport) {
+    dependsOn = [test, subprojects.jacocoTestReport]
+    additionalSourceDirs.setFrom files(subprojects.sourceSets.main.allSource.srcDirs)
+    sourceDirectories.setFrom files(subprojects.sourceSets.main.allSource.srcDirs)
+    classDirectories.setFrom files(subprojects.sourceSets.main.output)
+    executionData.setFrom project.fileTree(dir: '.', include: '**/build/jacoco/test.exec')
+    reports {
+        xml.setRequired true
+        csv.setRequired false
+        html.setRequired true
+    }
+}
+
+tasks.withType(JavaCompile).configureEach {
+    options.compilerArgs = [
+            '-Amapstruct.suppressGeneratorTimestamp=true'
+    ]
+}
+tasks.named('wrapper') {
+    gradleVersion = '8.1.1'
+}

--- a/src/test/resources/jacoco-merged/before.txt
+++ b/src/test/resources/jacoco-merged/before.txt
@@ -1,0 +1,68 @@
+allprojects {
+    repositories {
+        mavenCentral()
+        maven { url "https://repo.spring.io/snapshot" }
+        maven { url "https://repo.spring.io/milestone" }
+    }
+}
+
+subprojects {
+    apply plugin: 'java'
+    apply plugin: 'jacoco'
+    apply plugin: 'io.spring.dependency-management'
+
+    compileJava.dependsOn validateStructure
+    sourceCompatibility = JavaVersion.VERSION_17
+    test {
+        useJUnitPlatform()
+    }
+    dependencies {
+        testImplementation 'org.springframework.boot:spring-boot-starter-test'
+        compileOnly "org.projectlombok:lombok:${lombokVersion}"
+        annotationProcessor  "org.projectlombok:lombok:${lombokVersion}"
+        testCompileOnly  "org.projectlombok:lombok:${lombokVersion}"
+        testAnnotationProcessor  "org.projectlombok:lombok:${lombokVersion}"
+        implementation platform("org.springframework.boot:spring-boot-dependencies:${springBootVersion}")
+    }
+
+
+    test.finalizedBy(project.tasks.jacocoTestReport)
+
+    jacocoTestReport {
+        dependsOn test
+        reports {
+            xml.setRequired true
+            xml.setOutputLocation file("${buildDir}/reports/jacoco.xml")
+            csv.setRequired false
+            html.setOutputLocation file("${buildDir}/reports/jacocoHtml")
+        }
+    }
+
+}
+
+jacoco {
+    toolVersion = "${jacocoVersion}"
+    reportsDirectory.set(file("$buildDir/reports"))
+}
+
+tasks.register('jacocoMergedReport', JacocoReport) {
+    dependsOn = subprojects.jacocoTestReport
+    additionalSourceDirs.setFrom files(subprojects.sourceSets.main.allSource.srcDirs)
+    sourceDirectories.setFrom files(subprojects.sourceSets.main.allSource.srcDirs)
+    classDirectories.setFrom files(subprojects.sourceSets.main.output)
+    executionData.setFrom project.fileTree(dir: '.', include: '**/build/jacoco/test.exec')
+    reports {
+        xml.setRequired true
+        csv.setRequired false
+        html.setRequired true
+    }
+}
+
+tasks.withType(JavaCompile).configureEach {
+    options.compilerArgs = [
+            '-Amapstruct.suppressGeneratorTimestamp=true'
+    ]
+}
+tasks.named('wrapper') {
+    gradleVersion = '8.1.1'
+}

